meta_automl/surrogate/datasets/dataset_generate.py

```diff
@@ -0,0 +1,174 @@
+import sys
+sys.path.append("..")
+
+import pickle
+import shutil
+from tqdm import tqdm
+from typing import List, Union, Tuple, Dict, Optional, Any
+import json
+import pickle
+import os
+import openml
+import pandas as pd
+from sklearn.preprocessing import StandardScaler, OneHotEncoder
+
+from fedot.core.pipelines.pipeline import Pipeline
+
+from meta_automl.data_preparation.models_loaders import KnowledgeBaseModelsLoader
+from meta_automl.data_preparation.pipeline_features_extractors import FEDOTPipelineFeaturesExtractor
+from meta_automl.data_preparation.meta_features_extractors import OpenMLDatasetMetaFeaturesExtractor
+from meta_automl.data_preparation.feature_preprocessors import FeaturesPreprocessor
+from meta_automl.data_preparation.model import Model
+
+def calc_pipeline_hash(pl: Pipeline)-> Tuple[str]:
+    return tuple(str(pl.get_edges())+str(pl.nodes))
+
+
+class KnowledgeBaseToDataset:
+    def __init__(
+        self,
+        knowledge_base_directory: str,
+        dataset_directory: str,
+        meta_features_data_columns: List[str],
+        split: Optional[str] = "all", # Can be train, test, all
+        train_test_split_name: Optional[str] = "train_test_datasets_classification.csv",
+        task_type: Optional[str] = "classification",
+        fitness_metric: Optional[str] = "f1",
+        exclude_datasets: Optional[List[str]] = None,  # In accordance to @MorrisNein message in chat
```

Comment:
Если для внутреннего пользования больше не нужно:
```suggestion
        exclude_datasets: Optional[List[str]] = None
```